### PR TITLE
Update IKE 

### PR DESCRIPTION
**Pull request by allen0091**
- Add decrypt and encrypt
- Add sha256 algorithm
- Remove XFRM related code

### CODE REVIEW

**Comment by ianchen0119** [comment]
Please resolve the conflicts.

File: message/message.go [hunk 1]
```diff
@@ -8,15 +8,54 @@ import (
...
-	Payloads     IKEPayloadContainer
+	NextPayload  uint8
+}
+
+type IKEMessage struct {
+	IKEHeader
+	Payloads IKEPayloadContainer
+}
+
+func ParseIkeHeader(b []byte) (*IKEMessage, error) {
```
Comment (tim-ywliu) [inline message/message.go]: return (*IKEHeader, error)

File: ike.go [hunk 2]
```diff
@@ -29,25 +29,25 @@ func EncodeEncrypt(
 // and get IKESA
 func DecodeDecrypt(
 	msg []byte,
+	ikeMsg *message.IKEMessage,
```
Comment (tim-ywliu) [inline ike.go]: ikeHdr *IKEHeader

File: ike.go [hunk 3]
```diff
@@ -29,25 +29,25 @@ func EncodeEncrypt(
...
 	ikesaKey *security.IKESAKey,
 	role message.Role,
-) (*message.IKEMessage, error) {
-	ikeMsg := new(message.IKEMessage)
-	err := ikeMsg.Decode(msg)
+) error {
+	err := ikeMsg.Decode(msg[message.IKE_HEADER_LEN:])
```
Comment (tim-ywliu) [inline ike.go]:
```
ikeMsg := new(message.IKEMessage)
if ikeHdr == nil {
    err := ikeMsg.Decode(msg)
} else {
    ikeMsg.IKEHeader = ikeHdr
    err := ikeMsg.DecodePayload(msg[msg[message.IKE_HEADER_LEN:])
}
```

**Review by tim-ywliu — APPROVED** [review]

**Pull request merged**